packages/flutter/lib/src/material/popup_menu.dart

```diff
@@ -29,15 +29,49 @@ const double _kMenuVerticalPadding = 8.0;
 const double _kMenuWidthStep = 56.0;
 const double _kMenuScreenPadding = 8.0;
 
+/// A base class for entries in a material design popup menu.
+///
+/// The popup menu widget uses this interface to interact with the menu items.
+/// To show a popup menu, use the [showMenu] function. To create a button that
+/// shows a popup menu, consider using [PopupMenuButton].
+///
+/// See also:
+///
+///  * [PopupMenuItem]
+///  * [PopupMenuDivider]
+///  * [CheckedPopupMenuItem]
+///  * [showMenu]
+///  * [PopupMenuButton]
 abstract class PopupMenuEntry<T> extends StatefulWidget {
-  PopupMenuEntry({ Key key }) : super(key: key);
+  /// Abstract const constructor. This constructor enables subclasses to provide
+  /// const constructors so that they can be used in const expressions.
+  const PopupMenuEntry({ Key key }) : super(key: key);
 
+  /// The amount of vertical space occupied by this entry.
+  ///
+  /// This value must remain constant for a given instance.
   double get height;
+
+  /// The value that should be returned by [showMenu] when the user selects this entry.
   T get value => null;
+
+  /// Whether the user is permitted to select this entry.
   bool get enabled => true;
```

Comment:
maybe the base class shouldn't have a default value, in fact. Just make it abstract.